commands/mr/create/mr_create.go

```diff
@@ -509,3 +509,45 @@ func generateMRCompareURL(opts *CreateOpts, repo glrepo.Interface) (string, erro
 		targetProjectID)
 	return u.String(), nil
 }
+
+func resolvedHeadRepo(f *cmdutils.Factory) func() (glrepo.Interface, error) {
+	return func() (glrepo.Interface, error) {
+		httpClient, err := f.HttpClient()
+		if err != nil {
+			return nil, err
+		}
+		remotes, err := f.Remotes()
+		if err != nil {
+			return nil, err
+		}
+		repoContext, err := glrepo.ResolveRemotesToRepos(remotes, httpClient, "")
+		if err != nil {
+			return nil, err
+		}
+		headRepo, err := repoContext.HeadRepo(true)
+		if err != nil {
+			return nil, err
+		}
+
+		return headRepo, nil
+	}
+}
+
+func headRepoOverride(f *cmdutils.Factory, opts *CreateOpts, repo string) error {
+	opts.HeadRepo = func() (glrepo.Interface, error) {
+		return glrepo.FromFullName(repo)
+	}
+	newRepo, err := opts.HeadRepo()
+	if err != nil {
+		return err
+	}
+	// Initialise new http client for new repo host
+	cfg, err := f.Config()
+	if err == nil {
+		cmdutils.OverrideAPIProtocol(cfg, newRepo)
+	}
+	f.HttpClient = func() (*gitlab.Client, error) {
+		return cmdutils.LabClientFunc(newRepo.RepoHost(), cfg, false)
+	}
```

Comment:
The `headRepo` is basically for pushing branches so it has nothing to do with the `HttpClient`. All API calls will be made with the `baseRepo`

```suggestion
```

Reply:
fixed, thanks.